### ISSUE TITLE
[Feature]: Git should show changes to app settings

### ISSUE THREAD

**Issue by dhruvikn** [appsmithorg/appsmith#22213]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

TLDR -  When changing app settings, it is committing to git but changes are not shown in the commit modal. Not only for app navigation but for the entire app settings.

Check the loom for a detailed walkthrough.

### Steps To Reproduce

1. Change any of the app settings.
2. Click on the deploy button or the git commit button.
3. Observe how the heading `Changes since last deployment` shows up but the changes list is empty.

### Environment

Production

### Issue video log

https://lcixo7xrxt.vmaker.com/record/9pFJpUkXccgtMk2V

### Version

Can be observed on release [Appsmith v1.9.16-SNAPSHOT] and production [Appsmith v1.9.15-SNAPSHOT].

**Comment by AnaghHegde** [comment]
Its a front end issue i believe. This data is part of status API response.
